### ISSUE TITLE
Need to create Big Query Primary Key for Data Stream to work.

### ISSUE THREAD

**Issue by bradenwright** [hashicorp/terraform-provider-google#13323]
### Description

I can successfully create Big Query Tables and stream into them using Data Stream (so we can setup partitioning vs having datastream  automatically create the tables without partitioning, this medium article has a good example: [Link to Medium Article](https://medium.com/google-cloud/configure-streams-in-datastream-with-predefined-tables-in-bigquery-for-postgresql-as-source-528340f7989b) which has us create a table using a command like:

```
CREATE OR REPLACE TABLE  `projectname.partition_test.public_order_non_partition_onepkcolumn`
  (
  order_id INT64,
  ordered_at TIMESTAMP,
  shipped_at TIMESTAMP,
  order_amount FLOAT64,
  customer_id STRING(255),
  datastream_metadata STRUCT<uuid STRING, source_timestamp INT64>,
    PRIMARY KEY (order_id) NOT ENFORCED)
PARTITION BY
  DATE(ordered_at)
CLUSTER BY order_id
OPTIONS(max_staleness = INTERVAL 5 MINUTE);
```

I don't know of any way via Terraform to specific `PRIMARY KEY (order_id) NOT ENFORCED)` which causes datastream to throw the following error: 

```
jsonPayload: {
context: "CDC"
event_code: "UNSUPPORTED_EVENTS_DISCARDED"
message: "Discarded 3502 unsupported events for BigQuery destination: 833537404433.dev_everwash_dwh_internal.Membership, with reason code: BIGQUERY_UNSUPPORTED_PRIMARY_KEY_CHANGE, details: Failed to write to BigQuery due to an unsupported primary key change: adding primary keys to existing tables is not supported.."
object_name: "internal_Membership"
read_method: ""
}
```

### New or Affected Resource(s)

google_bigquery_table

### References

[Link to Medium Article](https://medium.com/google-cloud/configure-streams-in-datastream-with-predefined-tables-in-bigquery-for-postgresql-as-source-528340f7989b)

**Comment by bradenwright** [comment]
Any updates on this issue?  is it on the roadmap to add this feature?

**Comment by jllyfsh** [comment]
Keen to hear about this too

**Comment by KishinNextCompara** [comment]
if I understood the problem correctly, I think it is related to this [issue](https://issuetracker.google.com/issues/250938168?pli=1)  (comment #39) Datastream can't add partitioning/pk to an existing BigQuery table, if it's possible you should create all tables with partition before running Datastream.

**Comment by nilsjorgen** [comment]
>  if it's possible you should create all tables with partition before running Datastream.

Yes, but I belive the point of this issue is the ability to do exactly that, but with Terraform.

I have the exact same problem. Right now create Datastreams and all other related resources using Terraform, for for the tables I want to be partition, I have to create them with SQL first.
